core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -86,6 +126,21 @@ class InvokerPool(
 
         case GetStatus => sender() ! invokerStatus.toMap
 
+        case msg: InvocationFinishedMessage => {
+            // Forward message to invoker, if InvokerActor exists
+            invokers.get(msg.name).map(_.forward(msg))
+        }
+
+        case msg: ActivationRequest => {
```

Comment:
My 2 cents: I cannot quantify the impact. It is true, that the messages are serialized through the actor, but the kafka sending runs parallelized because the actor is not blocked by it. So the serialization could (should even?) be sufficiently small to be neglectable.